### PR TITLE
Added disableBundleAccessors config generation option

### PR DESCRIPTION
**Pull request by wojciech-kulik**
Resolves https://github.com/tuist/tuist/issues/3052 and https://github.com/tuist/tuist/discussions/3077

### Short description 📝

Added `disableBundleAccessors` config generation option.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] ~~In case the PR introduces changes that affect users, the documentation has been updated.~~

### CODE REVIEW

File: projects/tuist/fixtures/ios_app_with_framework_and_disabled_resources/StaticFramework/Sources/StaticFrameworkResouces.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Foundation
+import UIKit
+
+public class StaticFrameworkResources {
```
Comment (fortmarek) [inline projects/tuist/fixtures/ios_app_with_framework_and_disabled_resources/StaticFramework/Sources/StaticFrameworkResouces.swift:4]: nit: there's a typo in the name of the file

File: Sources/ProjectDescription/Config.swift [hunk 2]
```diff
@@ -128,6 +132,10 @@ extension Config.GenerationOptions {
             try container.decode(Bool.self, forKey: .resolveDependenciesWithSystemScm)
         {
             self = .resolveDependenciesWithSystemScm
+        } else if container.allKeys.contains(.disableBundleAccessors),
```
Comment (fortmarek) [inline Sources/ProjectDescription/Config.swift:135]: could you test this logic by adding this option [here](https://github.com/tuist/tuist/blob/main/Tests/ProjectDescriptionTests/ConfigTests.swift#L5)?

File: Tests/TuistKitTests/GraphMappers/ProjectMapperProviderTests.swift [hunk 3]
```diff
@@ -86,6 +86,24 @@ final class ProjectMapperProviderTests: TuistUnitTestCase {
         XCTAssertEqual(sequentialProjectMapper.mappers.filter { $0 is SynthesizedResourceInterfaceProjectMapper }.count, 0)
     }
 
+    func test_mappers_does_not_returns_resources_project_mapper_when_disabled_bundle_accessors() throws {
```
Comment (fortmarek) [inline Tests/TuistKitTests/GraphMappers/ProjectMapperProviderTests.swift:89]:
```suggestion
    func test_mappers_does_not_return_resources_project_mapper_when_disabled_bundle_accessors() throws {
```

File: projects/tuist/fixtures/ios_app_with_framework_and_disabled_resources/Tuist/Config.swift [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import ProjectDescription
+
+let config = Config(
+  generationOptions: [
+    .disableBundleAccessors,
```
Comment (fortmarek) [inline projects/tuist/fixtures/ios_app_with_framework_and_disabled_resources/Tuist/Config.swift:5]: thanks a lot for adding this fixture! It's very much appreciated.

However, I think it'd be great if you also ran some acceptance test for this to ensure this fixture is compilable. You can also assert that those derived files are not created.

You can add the acceptance test [here](https://github.com/tuist/tuist/blob/main/projects/tuist/features/generate-8.feature). There is no assertion at the moment for files not existing, but you can just more-or-less copy [this](https://github.com/tuist/tuist/blob/main/projects/tuist/features/step_definitions/generate.rb#L126) and negate it.

You can read more about acceptance tests [here](https://docs.tuist.io/contributors/testing-strategy)

**Review by fortmarek — COMMENTED** [review]
This is great, thanks @wojciech-kulik!

Could you also add this case to the website documentation [here](https://github.com/tuist/tuist/blob/main/projects/docs/docs/manifests/config.md)?

File: Sources/ProjectDescription/Config.swift (hunk 2 above)
Comment (wojciech-kulik) [reply Sources/ProjectDescription/Config.swift:135]: I've already added this option in `ConfigTests.swift` (https://github.com/tuist/tuist/pull/3088/files#diff-98305cf446c3f2af6eae9e4746a79d6671dde48b998ff784b42cd822621d4c73) What else would you like to have there?

Comment (fortmarek) [reply Sources/ProjectDescription/Config.swift:135]: ah, sorry, missed it, consider this as resolved then 🙂

**Comment by fortmarek** [comment]
@all-contributors add @wojciech-kulik for code

**Review by fortmarek — APPROVED** [review]
Well done @wojciech-kulik! 👏

**Comment by wojciech-kulik** [comment]
> Well done @wojciech-kulik! 👏

Thanks :)!

**Review by pepibumur — APPROVED** [review]

**Comment by pepibumur** [comment]
I think we can merge once CI is green :)

**Comment by messin9** [comment]
is this option still available in 3.0 version?

**Comment by ahmet-investa** [comment]
> is this option still available in 3.0 version?

You can use it as Project options.

`
let project = Project(
    name: "TestProj",
    options: Project.Options.options(disableBundleAccessors: true),

`

**Comment by ahmetws** [comment]
> is this option still available in 3.0 version?

You can use it as Project options.

`
let project = Project(
    name: "TestProj",
    options: Project.Options.options(disableBundleAccessors: true),

`

**Pull request merged**